public/javascripts/items.js

```diff
@@ -7,28 +7,105 @@ const fetchParams = {
 
-$(document).ready( () => {
-  $( '.title' ).click( event => {
-    const element = $( event.target )
-    const id = element.data( 'id' )
-    const completed = ! element.data( 'completed' )
+const RETURN_KEY = 13
+
+const checkJsonForSuccessField = json => {
+  if( json.success ) {
+    Promise.resolve( json )
+  } else {
+    Promise.reject( json.message )
+  }
+}
+
+const titleEdited = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+
+  if( event.charCode === RETURN_KEY ) {
+    let updatedTitle = element[0].value
+    fetch( `/items/${id}/edit`, params( { title: updatedTitle } ) )
+    .then( result => result.json() )
+    .then ( checkJsonForSuccessField )
+    .then( json => {
+      let span = $( `.title[data-id=${id}] > span`)
+      $( span[0] ).html(updatedTitle)
+      element.addClass( 'hidden')
+      $( span ).removeClass( 'hidden' )
+    })
+    .catch( error => {
+      console.log('error:', error)
+    })
+  }
+}
+
+const titleClicked = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+  $( element[0] ).addClass( 'hidden' )
+  let input = $( `.title[data-id=${id}] > input`)
+  $( input[0]).removeClass( 'hidden' )
+}
 
-    fetch( `/items/${id}/completed`, params( completed ))
+const descriptionEdited = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+
+  if( event.charCode === RETURN_KEY ) {
+    let updatedDescription = element[0].value
+    fetch( `/items/${id}/edit`, params( { description: updatedDescription } ) )
       .then( result => result.json() )
+      .then( checkJsonForSuccessField )
       .then( json => {
-        if( json.success ) {
-          element.data( 'completed', completed )
-
-          const parent = $( `.item[data-id=${id}]` )
-          if( completed ) {
-            parent.addClass( 'completed' )
-          } else {
-            parent.removeClass( 'completed' )
-          }
-        }
+        let span = $(`.description[data-id=${id}] > span`)
+        $( span[0] ).html( updatedDescription )
+        element.addClass( 'hidden')
+        $( span ).removeClass( 'hidden' )
+      })
+      .catch( error => {
+        console.log('error:', error)
       })
+  }
+}
+
+const descriptionClicked = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+  $( element[0] ).addClass( 'hidden' )
+  let input = $(`.description[data-id=${id}] > input`)
+  $( input[0]).removeClass( 'hidden' )
+}
+
+const completedClicked = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+  const completed = ! element.data( 'completed' )
+
+  fetch( `/items/${id}/completed`, params({ completed: completed } ) )
+  .then( result => result.json() )
+  .then( checkJsonForSuccessField )
+  .then( json => {
+      element.data( 'completed', completed )
+
+      const parent = $( `.title[data-id=${id}]` )
```

Comment:
Please place declarations at the top of their enclosing scope to aid in readability:
```js
.then( json => {
  const parent = $( `.title[data-id=${id}]` )

  element.data( 'completed', completed )
  if( completed ) { 
    // etc
  }
})
```